File: mycollab-scheduler-ondemand/src/main/scala/com/esofthead/mycollab/schedule/jobs/DeleteObsoleteLiveInstancesJob.scala
package com.esofthead.mycollab.schedule.jobs

import com.esofthead.mycollab.common.dao.LiveInstanceMapper
import com.esofthead.mycollab.common.domain.LiveInstanceExample
import org.joda.time.LocalDate
import org.quartz.{JobExecutionContext, JobExecutionException}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.config.BeanDefinition
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component

/**
  * @author MyCollab Ltd
  * @since 5.2.6
  */
@Component
@Scope(BeanDefinition.SCOPE_PROTOTYPE)
class DeleteObsoleteLiveInstancesJob extends GenericQuartzJobBean {
  @Autowired private val liveInstanceMapper: LiveInstanceMapper = null

  @throws(classOf[JobExecutionException])
  def executeJob(context: JobExecutionContext): Unit = {
    val ex = new LiveInstanceExample
    ex.createCriteria().andLastupdateddateLessThan(new LocalDate().minusDays(30).toDate)
    liveInstanceMapper.deleteByExample(ex)
  }
}

File: mycollab-scheduler-ondemand/src/main/java/com/mycollab/ondemand/schedule/jobs/DeleteObsoleteLiveInstancesJob.kt
package com.mycollab.ondemand.schedule.jobs

import com.mycollab.common.domain.LiveInstanceExample
import com.mycollab.pro.common.dao.LiveInstanceMapper
import com.mycollab.schedule.jobs.GenericQuartzJobBean
import org.quartz.JobExecutionContext
import org.quartz.JobExecutionException
import org.springframework.beans.factory.config.BeanDefinition
import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import java.time.LocalDateTime

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
@Component
@Scope(BeanDefinition.SCOPE_PROTOTYPE)
class DeleteObsoleteLiveInstancesJob(private val liveInstanceMapper: LiveInstanceMapper) : GenericQuartzJobBean() {

    @Throws(JobExecutionException::class)
    override fun executeJob(context: JobExecutionContext) {
        val ex = LiveInstanceExample()
        ex.createCriteria().andLastupdateddateLessThan(LocalDateTime.now().minusDays(1))
        liveInstanceMapper.deleteByExample(ex)
    }
}